ruby/ql/lib/codeql/ruby/frameworks/ActionView.qll

```diff
@@ -216,4 +216,138 @@ class FileSystemResolverAccess extends DataFlow::CallNode, FileSystemAccess::Ran
+  /**
+   * Action view helper methods which are XSS sinks.
+   */
+  module Helpers {
+    /**
+     * A call to an ActionView helper which renders its argument without escaping.
+     * The argument should be treated as an XSS sink. In the documentation for
+     * classes in this module, the vulnerable argument is named `x`.
+     */
+    abstract class RawHelperCall extends MethodCall {
+      /**
+       * Get an argument which is rendered without escaping.
+       */
+      abstract Expr getRawArgument();
+    }
+
+    /**
+     * `ActionView::Helpers::TextHelper#simple_format`.
+     *
+     * `simple_format(x, y, sanitize: false)`.
+     */
+    private class SimpleFormat extends ActionViewContextCall, RawHelperCall {
+      SimpleFormat() {
+        this.getMethodName() = "simple_format" and
+        this.getKeywordArgument("sanitize").getConstantValue().isBoolean(false)
+      }
+
+      override Expr getRawArgument() { result = this.getArgument(0) }
+    }
+
+    /**
+     * `ActionView::Helpers::TextHelper#truncate`.
+     *
+     * `truncate(x, escape: false)`.
+     */
+    private class Truncate extends ActionViewContextCall, RawHelperCall {
+      Truncate() {
+        this.getMethodName() = "truncate" and
+        this.getKeywordArgument("escape").getConstantValue().isBoolean(false)
+      }
+
+      override Expr getRawArgument() { result = this.getArgument(0) }
+    }
+
+    /**
+     * `ActionView::Helpers::TextHelper#highlight`.
+     *
+     * `highlight(x, y, sanitize: false)`.
+     */
+    private class Highlight extends ActionViewContextCall, RawHelperCall {
+      Highlight() {
+        this.getMethodName() = "highlight" and
+        this.getKeywordArgument("sanitize").getConstantValue().isBoolean(false)
+      }
+
+      override Expr getRawArgument() { result = this.getArgument(0) }
+    }
+
+    /**
+     * `ActionView::Helpers::JavascriptHelper#javascript_tag`.
+     *
+     * `javascript_tag(x)`.
+     */
+    private class JavascriptTag extends ActionViewContextCall, RawHelperCall {
+      JavascriptTag() { this.getMethodName() = "javascript_tag" }
+
+      override Expr getRawArgument() { result = this.getArgument(0) }
+    }
+
+    /**
+     * `ActionView::Helpers::TagHelper#tag`.
+     *
+     * `tag(x, x, y, false)`.
+     */
+    private class ContentTag extends ActionViewContextCall, RawHelperCall {
+      ContentTag() {
+        this.getMethodName() = "content_tag" and
+        this.getArgument(3).getConstantValue().isBoolean(false)
+      }
+
+      override Expr getRawArgument() { result = this.getArgument(1) }
+    }
+
+    /**
+     * `ActionView::Helpers::TagHelper#tag`.
+     *
+     * `tag(x, x, y, false)`.
+     */
+    private class Tag extends ActionViewContextCall, RawHelperCall {
+      Tag() {
+        this.getMethodName() = "tag" and
+        this.getArgument(3).getConstantValue().isBoolean(false)
+      }
+
+      override Expr getRawArgument() { result = this.getArgument(0) }
```

Comment:
It seems like the `false` argument to `escape` disables escaping of attribute values, so this should maybe find keyword argument values instead? I used the `Legacy syntax` section of https://api.rubyonrails.org/v5.1/classes/ActionView/Helpers/TagHelper.html#method-i-tag as reference.

Reply:
Interesting, thanks for testing that.